php/src/Communicator.cpp

```diff
@@ -1155,9 +943,25 @@ ZEND_FUNCTION(Ice_initialize)
                 }
             }
         }
-    }
 
-    initData.valueFactoryManager = make_shared<ValueFactoryManager>();
+        member = "sliceLoader";
+        {
+            if ((data = zend_hash_str_find(Z_OBJPROP_P(zvinit), member.c_str(), member.size())) != 0)
+            {
+                assert(Z_TYPE_P(data) == IS_INDIRECT);
+                initDataSliceLoader = Z_INDIRECT_P(data);
+
+                if (!ZVAL_IS_NULL(initDataSliceLoader))
+                {
+                    if (Z_TYPE_P(initDataSliceLoader) != IS_OBJECT)
```

Comment:
I could not find a way to check the type implements interface SliceLoader. But it's not need now that the type enforcement is performed at the PHP level.

Reply:
> But it's not need now that the type enforcement is performed at the PHP level.

Why do we need to check it is an object then?

Reply:
I'll remove it.